minet/bluesky/http_client.py

```diff
@@ -101,30 +110,136 @@ def request(
             headers=headers,
         )
 
+        if response.status < 200 or response.status >= 300:
+            data = response.json()
+            if "error" in data:
+                e = data["error"]
+                if response.status == 400:
+                    if e == "UpstreamFailure":
+                        raise BlueskyUpstreamFailureError(
+                            "Bluesky is currently experiencing upstream issues."
+                        )
+                    elif e in (
+                        "BadRequest",
+                        "InvalidRequest",
+                        "must pass non-empty search query",
+                    ):
+                        raise BlueskyBadRequestError(data["message"])
+                raise BlueskyError(f"{e}: {data['message']}")
```

Comment:
Avoid using a non-specific `BlueskyError`. Maybe this and/or the following can be a `BlueskyRequestError`.